src/EditorFeatures/Test2/FindReferences/FindReferencesTests.Tuples.vb

```diff
@@ -409,6 +409,198 @@ partial class Program
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferencesNetCoreApp="true">
+        <Document><![CDATA[
+using System;
+
+class C
+{
+    (int, int) M()
+    {
+        return (1, 1);
+    }
+
+    void M2()
+    {
+        int {|Definition:x|};
+        int y;
+
+        ($$[|x|], y) = M();
+
+        [|x|] = 0;
+    }
+}
+]]>
+        </Document>
+    </Project>
+</Workspace>
+            Await TestAPIAndFeature(input, kind, host)
+        End Function
+
+        <WorkItem("https://github.com/dotnet/roslyn/issues/52621")>
+        <WpfTheory, CombinatorialData>
+        Public Async Function TestTupleSwappedFields(kind As TestKind, host As TestHost) As Task
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferencesNetCoreApp="true">
+        <Document><![CDATA[
+using System;
+
+class C
+{
+    void M(int {|Definition:left|}, int right)
+    {
+        var r = ($$[|left|], right);
+        r = (right, [|left|]);
+    }
+}
+]]>
+        </Document>
+    </Project>
+</Workspace>
+            Await TestAPIAndFeature(input, kind, host)
+        End Function
+
+        <WorkItem("https://github.com/dotnet/roslyn/issues/52621")>
+        <WpfTheory, CombinatorialData>
+        Public Async Function TestTupleNonLocal(kind As TestKind, host As TestHost) As Task
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferencesNetCoreApp="true">
+        <Document><![CDATA[
+using System;
+
+class C
+{
+    int {|Definition:Property|} { get; set; }
+
+    void M(string a)
+    {
+        var r = (a.Length, $$[|Property|]);
+        r = ([|Property|], Property: [|Property|]);
+    }
+}
+]]>
+        </Document>
+    </Project>
+</Workspace>
+            Await TestAPIAndFeature(input, kind, host)
+        End Function
+
+        <WorkItem("https://github.com/dotnet/roslyn/issues/52621")>
+        <WpfTheory, CombinatorialData>
+        Public Async Function TestTupleExplicitNames(kind As TestKind, host As TestHost) As Task
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferencesNetCoreApp="true">
+        <Document><![CDATA[
+using System;
+
+class C
+{
+    void M(int a, int b)
+    {
+        var t = ($$[|{|Definition:x|}|]: a, y: b);
+        t = ([|x|]: b, y: a);
+        b = t.[|x|];
+    }
+}
+]]>
+        </Document>
+    </Project>
+</Workspace>
+            Await TestAPIAndFeature(input, kind, host)
+        End Function
+
+        <WorkItem("https://github.com/dotnet/roslyn/issues/52621")>
+        <WpfTheory, CombinatorialData>
+        Public Async Function TestTupleExplicitNamesSameAsLocals(kind As TestKind, host As TestHost) As Task
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferencesNetCoreApp="true">
+        <Document><![CDATA[
+using System;
+
+class C
+{
+    void M(int x, int y)
+    {
+        var t = ($$[|{|Definition:x|}|]: x, y: y);
+        t = ([|x|]: y, y: x);
```

Comment:
test of `x:x` as well please.

Reply:
`x: x` is included above, it's the definition, do you also want a test that doesn't involve the definition itself?

Reply:
yes.

Reply:
add test where there is also an implicit name.  i.e. `(x, ...)`